ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2

```diff
@@ -1379,6 +1379,18 @@ router bgp {{ router_bgp.as }}
 {%         endif %}
 {%         if vrf.evpn_multicast is arista.avd.defined(true) %}
       evpn multicast
+{%             if vrf.evpn_multicast_gateway_dr_election_algorithm.hrw is arista.avd.defined(true) or vrf.evpn_multicast_gateway_dr_election_algorithm.modulus is arista.avd.defined(true)
+                   or vrf.evpn_multicast_gateway_dr_election_algorithm.preference is arista.avd.defined %}
+{%                 set election_cli = "gateway dr election algorithm" %}
+{%                 if vrf.evpn_multicast_gateway_dr_election_algorithm.hrw is arista.avd.defined(true) %}
+{%                     set election_cli = election_cli ~ " hrw" %}
+{%                 elif vrf.evpn_multicast_gateway_dr_election_algorithm.modulus is arista.avd.defined(true) %}
+{%                     set election_cli = election_cli ~ " modulus" %}
+{%                 else %}
+{%                     set election_cli = election_cli ~ " preference" ~ " " ~ vrf.evpn_multicast_gateway_dr_election_algorithm.preference %}
+{%                 endif %}
+         {{ election_cli }}
```

Comment:
We need ! between dr electtion command and address family command
```
router bgp 65000
   vrf data
      evpn multicast
         gateway dr election algorithm modulus
         !
         address-family ipv4
```

Reply:
Also i noticed that when we do the dr election algorithm enabled it generate the  `address-family ipv4` config too.

Reply:
We have seperate key vrf.evpn_multicast_address_family.ipv4 for address-family-ipv4 configuration.

Reply:
should we also enable `address-family ipv4` config when we enable `dr election algorithm` ?
@ClausHolbechArista

Reply:
No eos_cli_config_gen should not be taking any decisions on config. Just print what is given as long as it is valid.

Reply:
We probably need a knob for address_family_ipv4 (dict) under the evpn_multicast context. This should have an `enabled` boolean. If true just print only the header. If false print the header and "disabled".

Reply:
{%             if vrf.evpn_multicast_address_family.ipv4 is arista.avd.defined
                   and vrf.evpn_multicast_address_family.ipv4.transit is arista.avd.defined(true) %}
         address-family ipv4
{%                 if vrf.evpn_multicast_address_family.ipv4.transit is arista.avd.defined(true) %}
            transit
{%                 endif %}

we already have this condition for address-family-ipv4. Does this work?